sharktank/sharktank/ops/default_impls.py

```diff
@@ -141,36 +141,19 @@ def elementwise_binary(operator, x, y, *args, **kwargs):
 
 
 @elementwise.override(
-    AllOfExprsVariadic(
-        IsOfType(Tensor, InferenceTensor),
-        IsOfType(Tensor, InferenceTensor, Number),
-        IsOfType(Tensor, InferenceTensor, Number),
+    AllOfExprs(
+        IsOfType(Tensor, PrimitiveTensor),
+        IsOfType(Tensor, PrimitiveTensor, Number),
+        IsOfType(Tensor, PrimitiveTensor, Number),
     )
 )
-def elementwise_variadic(operator, x, y, *args):
-    """Folds by successively applying the binary operator from left to right until
-    exhaustion.
-
-    Match a variable number of tensor/number arguments with at least 3 such arguments.
-
-    Example matches
-    ```
-    (Tensor, Tensor, Tensor)
-    (Tensor, DefaultPrimitiveTensor, float),
-    (SplitPrimitiveTensor, ReplicatedTensor, int, Tensor)
-    ```
-
-    Will not match
-    ```
-    (Tensor)
-    (Tensor, Tensor)
-    (int, Tensor, Tensor)
-    ```
-    """
-    res = elementwise(operator, x, y)
-    for arg in args:
-        res = elementwise(operator, res, arg)
-    return res
+def elementwise_trenary(operator, x, y, z, *args, **kwargs):
```

Comment:
Why is this change needed? Also should this be elementwise_ternary?

Reply:
I also expect this should be InferenceTensor

Reply:
Fixed the typo.
Originally I added this for convenience. Here is removed as it conflicts with dispatching to ternary ops. If the user wants binary op + folding they should do that themselves.

Reply:
@KyleHerndon, I think this should not accept any inference tensor as we do unboxing, which would be undesirable for some tensor types. The unboxing here is fine as it is a no-op.
If someone wants an implementation for example for quantized tensors they should write one.
I guess it is about whether we want to perform potentially a very inefficient computation or to fail.